src/Model/Resolver/ResetPassword.php

```diff
@@ -36,12 +38,25 @@ class ResetPassword implements ResolverInterface {
      */
     protected $session;
 
+    /**
+     * @var ConfirmCustomerByToken
+     */
+    protected ConfirmCustomerByToken $confirmByToken;
+
+    /**
+     * @param Session $customerSession
+     * @param AccountManagementInterface $accountManagement
+     * @param ConfirmCustomerByToken|null $confirmByToken
+     */
     public function __construct(
         Session $customerSession,
-        AccountManagementInterface $accountManagement
+        AccountManagementInterface $accountManagement,
+        ConfirmCustomerByToken $confirmByToken = null
     ) {
         $this->session = $customerSession;
         $this->accountManagement = $accountManagement;
+        $this->confirmByToken = $confirmByToken
+            ?? ObjectManager::getInstance()->get(ConfirmCustomerByToken::class);
```

Comment:
Can `$this->confirmByToken` ever be `null`?

Reply:
I did not get the value as null in this resolver, but it is used this way everywhere in default Magento

Reply:
The code after the `??` operator will never be called if you run `magento setup:di:compile`.
Please remove it.